core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -1100,8 +1115,12 @@ private[spark] class TaskSetManager(
 
   def executorDecommission(execId: String): Unit = {
     recomputeLocality()
-    // Future consideration: if an executor is decommissioned it may make sense to add the current
-    // tasks to the spec exec queue.
+    if (executorDecommissionKillInterval.nonEmpty) {
+      val executorKillTime = clock.getTimeMillis() + executorDecommissionKillInterval.get
+      runningTasksSet.filter(taskInfos(_).executorId == execId).foreach { tid =>
+        tidToExecutorKillTimeMapping(tid) = executorKillTime
+      }
+    }
```

Comment:
nit:

```suggestion
    executorDecommissionKillInterval.foreach { interval =>
      val executorKillTime = clock.getTimeMillis() + interval
      taskInfos.filter(info.executorId == execId && info.running).foreach { info =>
        tidToExecutorKillTimeMapping(into.taskId) = executorKillTime
      }
    }
```

Reply:
1. Added `executorDecommissionKillInterval.foreach { interval =>` to remove the if check.
2. Its better to filter runningTasksSet as compared to taskInfos as it is going to be relatively small.